### PR TITLE
Sorbet VS Code extension

### PR DESCRIPTION
**Pull request by jvilk-stripe**
### Motivation

Sorbet VS Code extension.

### Test plan

See included automated tests.

### CODE REVIEW

File: vscode_extension/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Version history
+
+## 0.3.6
+
```
Comment (jvilk-stripe) [inline vscode_extension/CHANGELOG.md:4]: I think we can erase the changelog for earlier versions and just write "- First open source release" under this header.

File: vscode_extension/package.json [hunk 2]
```diff
@@ -0,0 +1,314 @@
+{
+  "name": "sorbet-vscode-extension",
+  "displayName": "Ruby Sorbet",
+  "description": "Ruby IDE features, powered by Sorbet.",
+  "author": "#sorbet-extension",
+  "license": "UNLICENSED",
```
Comment (jvilk-stripe) [inline vscode_extension/package.json:6]: We should specify a license.

File: vscode_extension/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+	"compilerOptions": {
+		"noUnusedLocals": true,
```
Comment (jvilk-stripe) [inline vscode_extension/tsconfig.json:3]: We should fix the tabbing in this file.

File: .buildkite/test-vscode-extension.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -euo pipefail
+
+apt-get update
+apt-get install -y libxshmfence-dev libnss3-dev libatk1.0-0 libatk-bridge2.0-0 \
+ libdrm2 xvfb libgdk-pixbuf2.0-0 libgtk-3-0 libgbm1	libasound2
```
Comment (jez) [inline .buildkite/test-vscode-extension.sh:7]:
```suggestion
# TODO: We probably want to move this into https://github.com/sorbet/sorbet-build-image eventually

apt-get update
apt-get install -y libxshmfence-dev libnss3-dev libatk1.0-0 libatk-bridge2.0-0 \
 libdrm2 xvfb libgdk-pixbuf2.0-0 libgtk-3-0 libgbm1	libasound2
```

**Review by jez — APPROVED** [review]

File: .buildkite/pipeline.yaml [hunk 5]
```diff
@@ -80,6 +82,13 @@ steps:
     command: .buildkite/build-emscripten.sh
     artifact_paths: _out_/**/*
     <<: *elastic
+
+  - label: ":mac: build-vscode-extension.sh"
```
Comment (bardec) [inline .buildkite/pipeline.yaml:86]: It looks like I might have missed updating this to run on the linux agent :( 

Honestly we might not even need this right now until automatic publishing (I believe this was my intention originally). I'd recommend pulling it out, as the test should build it.

Comment (jvilk-stripe) [reply .buildkite/pipeline.yaml]: Done.

**Pull request merged**